docs/async/async-in-depth.md

```diff
@@ -0,0 +1,151 @@
+
+Although the above may seem like a lot of work to be done, when measured in terms of wall clock time, it’s miniscule compared to the time it takes to do the actual I/O work. Although not at all precise, a potential timeline for such a call would look like this:
+
+0-1————————————————————————————————————————————————–2-3
+
+*   Time spent from points `0` to `1` is everything up until an async method yields control to its caller.
+*   Time spent from points `1` to `2` is the time spent on I/O, with no CPU cost.
+*   Finally, time spent from points `2` to `3` is passing control back (and potentially a value) to the async method, at which point it is executing again.
+
+### What does this mean for a server scenario?
+
+This model works well with a typical server scenario workload.  Because tasks don't have an affinity to threads, it means that the server threadpool can use any thread for any task. As a result, it can service a much higher volume of web requests than if each thread were dedicated to running a particular request.
+
+Consider two servers: one that runs async code, and one that does not.  For the purpose of this example, each server only has 5 threads available to service requests.  Note that these numbers are imaginarily small and serve only in a demonstrative context.
+
+Assume both servers recieve 6 concurrent requests. Each request performs an I/O operation.  The server *without* async code has to queue up the 6th request until one of the 5 threads have finished the I/O-bound work and written a response. At the point that the 20th request comes in, the server might start to slow down, because the queue is getting too long.
+
+![](../../images/async/webserver-1.jpg)
+(caption) Web server with no async.
+
+As you can see, server threads can't service any other requests until the I/O they're assigned to start is finished.  That's not an ideal scenario!
+
+The server *with* async code running on it still queues up the 6th request, but because it uses `async` and `await`, each of its threads are freed up when the I/O-bound work starts, rather than when it finishes.  By the time the 20th request comes in, the queue for incoming requests will be far smaller (if it has anything in it at all), and the server won't slow down.
+
+![](../../images/async/webserver-2.jpg)
+(caption) Web server with async.
+
+As you can see, server threads are free to service more requests once they've started I/O.  The result of I/O gets queued up and the next available thread will handle writing the response for the result of the I/O.
```

Comment:
It might be worth mentioning that *any* of the threads can 'pick up' the response when the I/O operation completes, and return the result to the client. Even if the thread where the request started is busy on a new request, as long as any thread is available, work continues. They are all Threadpool threads.